include/System/BaseParam.hpp

```diff
@@ -3,12 +3,15 @@
 
 #include <JSystem/JSupport/JSUMemoryInputStream.hpp>
 
+class TParams;
+
 class TBaseParam {
 public:
-	virtual void load(JSUMemoryInputStream&);
+	TBaseParam(TParams* params, unsigned short code, const char* paramName);
+	virtual inline void load(JSUMemoryInputStream&);
```

Comment:
Na, this is wrong, just define the function here, inside of the class. When you do non-out-of-line method definitions, the method automatically gets inline linkage and so the symbol becomes weak. But as this symbol cannot be completely optimized out as it needs to be referenced by the vtable, it will get emitted into the first TU that has a definition for any other method of the class, which in this case is precisely BaseParam.cpp